arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentTableViewController.swift

```diff
@@ -14,40 +14,74 @@
 
 import UIKit
 
-class ContentTableViewController: UITableViewController, CustomSearchHeaderViewDelegate, DownloadProgressViewDelegate {
+class ContentTableViewController: UITableViewController {
+    
     var samples = [Sample]()
     private var expandedRowIndex:Int = -1
     
-    private var headerView:CustomSearchHeaderView!
     var containsSearchResults = false
     private var bundleResourceRequest:NSBundleResourceRequest!
     private var downloadProgressView:DownloadProgressView!
-
-    var token: Int = 0
+    
+    var searchEngine: SearchEngine?
+    
+    // strong reference needed for iOS 10
+    var filterSearchController:UISearchController?
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        self.tableView.rowHeight = UITableView.automaticDimension
-        self.tableView.estimatedRowHeight = 60
-        
-        if containsSearchResults {
-            self.tableView.tableHeaderView?.removeFromSuperview()
-            self.tableView.tableHeaderView = nil
-        }
-        else {
-            self.headerView = self.tableView.tableHeaderView! as? CustomSearchHeaderView
-            self.headerView.delegate = self
-            self.headerView.hideSuggestionsTable()
-        }
+        searchEngine = SearchEngine(samples: samples)
```

Comment:
`samples` may or may not be set at this point. It most likely is, but that could change in the future if the caller does something to trigger the view to load. A better way might be to use a property observer of `samples`:
```swift
var samples = [Sample]() {
   didSet {
      searchEngine = SearchEngine(samples: samples)
   }
}
```